### PR TITLE
Fix incorrect AbstractClassInstantiationHintError #6119

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/6119
- Check whether the class name has `$`
- Add unit tests

Before:

nb-php-gh-6119-abstract-instantiation-hint-error-before

After:

nb-php-gh-6119-abstract-instantiation-hint-error-after

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/AbstractClassInstantiationHintError.java [hunk 1]
```diff
@@ -110,7 +115,19 @@ public void visit(ClassInstanceCreation node) {
                 }
             }
         }
+    }
 
+    private boolean isDollaredName(ClassName className) {
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/verification/AbstractClassInstantiationHintError.java:120]: Don't we have already this method somewhere?

**Review by tmysik — APPROVED** [review]

File: php/php.editor/src/org/netbeans/modules/php/editor/verification/AbstractClassInstantiationHintError.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/AbstractClassInstantiationHintError.java:120]: No, It seems that we don't have it. (I searched "isDollared".)
I'll move it to `CodeUtils`.
Thanks!

Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/verification/AbstractClassInstantiationHintError.java]: Moved it.

**Comment by junichi11** [comment]
@tmysik Thank you for your reviews and your time :)

**Pull request merged**